.github/workflows/test-and-build.yml

```diff
@@ -67,6 +67,10 @@ jobs:
         CGO_ENABLED: 1
         LD_LIBRARY_PATH: ${{ github.workspace }}/candle-binding/target/release
 
+    - name: build container and run docker-compose

```

Comment:
why does the CI run docker compose?

Reply:
Just to ensure that providing docker-compose makes it easy for people to get started, perhaps it should be placed separately in an action job.  WDYT?

Reply:
Would you like to start a separate GHA with this so we can use it for integration with envoy?